lineapy/graph_reader/apis.py

```diff
@@ -159,7 +159,19 @@ def replace_fun(match):
 
             swapped, replaces = lineapy_pattern.subn(replace_fun, code)
             if replaces > 0:
+                # If we replaced something, pickle was used so add import pickle on top
+                # Conversely, if lineapy reference was removed, potentially the import lineapy line is not needed anymore.
+                remove_pattern = re.compile(r"import lineapy\n")
```

Comment:
The end line ensures that patterns like "import lineapy as lp" do not match. We do not handle aliases in the regex replace above so these would be out of scope for now until a better (potentially non-regexy) solution surfaces.